Extension/src/LanguageServer/colorization.ts

```diff
@@ -0,0 +1,601 @@
+        let otherSettings: OtherSettings = new OtherSettings(this.uri);
+
+        // Next in priority would be a global user override of token color of the equivilent scope
+        let colorTokenName: string | undefined = ColorizationSettings.scopeToTokenColorNameMap.get(scope);
+        if (colorTokenName) {
+            let settingValue: string = otherSettings.getCustomColorToken(colorTokenName);
+            if (settingValue) {
+                if (baseCStyle) {
+                    baseCStyle.foreground = settingValue;
+                }
+                if (baseCppStyle) {
+                    baseCppStyle.foreground = settingValue;
+                }
+            }
+        }
+
+        // Next in priority would be a global user override of this scope in textMateRules
+        this.findThemeStyleForScope(baseCStyle, baseCppStyle, scope, otherSettings.customTextMateRules);
+
+        // Next in priority would be a theme-specific user override of token color of the equivilent scope
+        if (colorTokenName) {
+            let settingValue: string = otherSettings.getCustomThemeSpecificColorToken(colorTokenName, themeName);
+            if (settingValue) {
+                if (baseCStyle) {
+                    baseCStyle.foreground = settingValue;
+                }
+                if (baseCppStyle) {
+                    baseCppStyle.foreground = settingValue;
+                }
+            }
+        }
+
+        // Next in priority would be a theme-specific user override of this scope in textMateRules
+        let textMateRules: TextMateRule[] = otherSettings.getCustomThemeSpecificTextMateRules(themeName);
+        this.findThemeStyleForScope(baseCStyle, baseCppStyle, scope, textMateRules);
+    }
+
+    private calculateThemeStyleForVsToken(tokenKind: TokenKind, scope: string, themeName: string, themeTextMateRules: TextMateRule[]): void {
+        let parts: string[] = scope.split(".");
+        let accumulatedScope: string = "";
+        for (let i: number = 0; i < parts.length; i++) {
+            accumulatedScope += parts[i];
+            this.calculateThemeStyleForScope(this.themeStyleCMap[tokenKind], this.themeStyleCppMap[tokenKind], accumulatedScope, themeName, themeTextMateRules);
+            this.calculateThemeStyleForScope(this.themeStyleCMap[tokenKind], null, accumulatedScope + ".c", themeName, themeTextMateRules);
+            this.calculateThemeStyleForScope(null, this.themeStyleCppMap[tokenKind], accumulatedScope + ".cpp", themeName, themeTextMateRules);
+            accumulatedScope += ".";
+        }
+    }
+
+    public reload(): void {
+        let otherSettings: OtherSettings = new OtherSettings(this.uri);
+        let themeName: string = otherSettings.colorTheme;
+        
+        // Enumerate through all extensions, looking for this theme.  (Themes are implemented as extensions - even the default ones)
+        // Crack open each package.json to check for a theme name patch
```

Comment:
This is open source code/comment. I would make comments as simple and clear as possible. I would remove "crack" and just say "open".